src/jemalloc/include/jemalloc/internal/jemalloc_internal.h.in

```diff
@@ -1041,7 +1041,7 @@ iralloct(void *ptr, size_t size, size_t extra, size_t alignment, bool zero,
 			}
 		}
 
-		return NULL;
+		return (ret != NULL) ? (ret) : (ptr);
```

Comment:
But it doesn't fail... :-)
Reallocation is successful - it returns the valid pointer to the allocation that is not smaller than requested. Does it violate the spec? It's perfectly fine if the pointer to the new object has the same value as to the old object), and I can't find any restrictions on the actual size of the object.

Note that this trick is used ONLY when the program attempts to shrink the allocation and (because of how the Jemalloc works and because of the limited size of the heap) it cannot be done.
It's not surprising if you can't reallocate from 16B to 16M and got OOM error. But if you can't reallocate from 16M to 16B and got OOM this is... not expected behavior.